python/gtsam/examples/TranslationAveragingExample.py

```diff
@@ -0,0 +1,122 @@
+    # Add the last two rotations.
+    rotations.insert(len(poses) - 1, poses[-1].rotation())
+    rotations.insert(len(poses) - 2, poses[-2].rotation())
+    return (rotations, translation_directions)
+
+
+def estimate_poses(relative_translations: gtsam.BinaryMeasurementsUnit3,
+                   rotations: gtsam.Values) -> gtsam.Values:
+    """Estimate poses given rotations normalized translation directions between cameras.
+
+    Args:
+        relative_translations -- List of normalized translation directions between camera pairs, each direction
+                                 is from the first camera to the second, in the frame of the first camera.
+        rotations -- Rotations of the cameras in the world frame.
+
+    Returns:
+        Values -- Estimated poses.
+    """
+
+    # Some hyperparameters, values used from 1dsfm.
+    max_1dsfm_projection_directions = 48
+    outlier_weight_threshold = 0.1
+
+    # Convert the translation directions to global frame using the rotations.
+    w_relative_translations = gtsam.BinaryMeasurementsUnit3()
+    for relative_translation in relative_translations:
+        w_relative_translation = gtsam.Unit3(rotations.atRot3(relative_translation.key1())
+                                             .rotate(relative_translation.measured().point3()))
+        w_relative_translations.append(gtsam.BinaryMeasurementUnit3(relative_translation.key1(),
+                                                                    relative_translation.key2(),
+                                                                    w_relative_translation,
+                                                                    relative_translation.noiseModel()))
+
+    # Indices of measurements that are to be used as projection directions. These are randomly chosen.
+    sampled_indices = np.random.choice(len(w_relative_translations), min(
+        max_1dsfm_projection_directions, len(w_relative_translations)), replace=False)
+    # Sample projection directions from the measurements.
+    projection_directions = [
+        w_relative_translations[idx].measured() for idx in sampled_indices]
+
+    outlier_weights = []
+    # Find the outlier weights for each direction using MFAS.
+    for direction in projection_directions:
+        algorithm = gtsam.MFAS(w_relative_translations, direction)
+        outlier_weights.append(algorithm.computeOutlierWeights())
+
+    # Compute average of outlier weights. Each outlier weight is a map from a pair of Keys (camera IDs) to a weight,
+    # where weights are proportional to the probability of the edge being an outlier.
+    avg_outlier_weights = defaultdict(lambda: 0.0)
+    for outlier_weight_dict in outlier_weights:
+        for keypair, weight in outlier_weight_dict.items():
+            avg_outlier_weights[keypair] += weight / len(outlier_weights)
+
+    # Remove w_relative_tranlsations that have weight greater than threshold, these are outliers.
```

Comment:
typo on `tranlsations`

Reply:
what does keypair mean?

Reply:
Keypair is actually a tuple (i, j), a pair of camera indices. I called it keypair because that is what the underlying C++ structure is called.

Reply:
Got it. thanks for the clarification. We still have a typo `w_relative_tranlsations`, and that variable should be renamed in the comment too